src/forms/numberInput.test.jsx

```diff
@@ -0,0 +1,179 @@
+		numberInputComponent = null;
+		inputEl = null;
+	});
+
+	it('exists', () => {
+		expect(() => TestUtils.findRenderedDOMComponentWithTag(numberInputComponent, 'input')).not.toThrow();
+	});
+
+	it('should have a name attribute', () => {
+		expect(inputEl.name).toEqual(NAME_ATTR);
+	});
+
+	it('should have a value when one is specified', () => {
+		expect(parseFloat(inputEl.value)).toEqual(VALUE);
+	});
+
+	it('should have a label when label is given', () => {
+		expect(() => TestUtils.findRenderedDOMComponentWithTag(numberInputComponent, 'label')).not.toThrow();
+		const labelEl = TestUtils.findRenderedDOMComponentWithTag(numberInputComponent, 'label');
+		expect(labelEl.textContent).toEqual(LABEL_TEXT);
+	});
+
+	it('should have a required attribute when specified', () => {
+		expect(inputEl.attributes.required).not.toBeNull();
+	});
+
+	it('should have a max attribute', () => {
+		expect(parseFloat(inputEl.max)).toEqual(MAX_ATTR);
+	});
+
+	it('should have a min attribute', () => {
+		expect(parseFloat(inputEl.min)).toEqual(MIN_ATTR);
+	});
+
+	it('should have a step attribute', () => {
+		expect(parseFloat(inputEl.step)).toEqual(STEP_ATTR);
+	});
+
+	it('should specify attributes that are passed in', function() {
+		const boundComponent = TestUtils.renderIntoDocument(
+			<NumberInput
+				name={NAME_ATTR}
+				label={LABEL_TEXT}
+				value={VALUE}
+				onChange={onChange}
+				placeholder={PLACEHOLDER}
+			/>
+		);
+		inputEl = TestUtils.findRenderedDOMComponentWithTag(boundComponent, 'input');
+
+		expect(inputEl.getAttribute('placeholder')).toEqual(PLACEHOLDER);
+	});
+
+	it('should have an error when one is specified', function() {
+		expect(() => TestUtils.findRenderedDOMComponentWithClass(numberInputComponent, 'text--error')).not.toThrow();
+		const errorEl = TestUtils.findRenderedDOMComponentWithClass(numberInputComponent, 'text--error');
+		expect(errorEl.textContent).toEqual(ERROR_TEXT);
+	});
+
+	it('should set its value on input change', function() {
+		const newValue = new Number(VALUE) + new Number(STEP_ATTR);
+		expect(parseFloat(inputEl.value)).toEqual(VALUE);
+		TestUtils.Simulate.change(inputEl, { target: { value: newValue } });
+		expect(parseFloat(inputEl.value)).toEqual(newValue);
+	});
+
+	it('should call onChange and setState with input change', function() {
+		const newValue = new Number(VALUE) + new Number(STEP_ATTR);
+		const changeSpy = spyOn(NumberInput.prototype, 'onChange').and.callThrough();
+		const stateSpy = spyOn(NumberInput.prototype, 'setState').and.callThrough();
+
+		const boundComponent = TestUtils.renderIntoDocument(
+			<NumberInput
+				name={NAME_ATTR}
+				label={LABEL_TEXT}
+				value={VALUE}
+				onChange={onChange}
+			/>
+		);
+
+		inputEl = TestUtils.findRenderedDOMComponentWithTag(boundComponent, 'input');
+		TestUtils.Simulate.change(inputEl, { target: { value: newValue } });
+
+		expect(changeSpy).toHaveBeenCalled();
+		expect(stateSpy).toHaveBeenCalledWith({ value: newValue });
```

Comment:
Since we are updating the setState to be a function we can change this test to be
`expect(numberInput.state.value).toEqual(newValue);`